virttest/test_setup.py

```diff
@@ -619,20 +619,20 @@ def set_hugepages(self):
         error_context.context(
             "setting hugepages limit to %s" % self.target_hugepages)
         try:
-            hugepage_cfg = open(self.kernel_hp_file, "r+")
-            hp = hugepage_cfg.readline().strip()
-            hugepage_cfg.close()
+            with open(self.kernel_hp_file, "r") as hugepage_cfg:
+                hp = hugepage_cfg.readline().strip()
         except IOError:
             raise exceptions.TestSetupFail("Can't read kernel hugepage file")
         while int(hp) < self.target_hugepages:
             loop_hp = hp
             try:
-                hugepage_cfg = open(self.kernel_hp_file, "r+")
-                hugepage_cfg.write(str(self.target_hugepages))
-                hugepage_cfg.flush()
-                hugepage_cfg.seek(0)
-                hp = int(hugepage_cfg.readline().strip())
-                hugepage_cfg.close()
+                with open("/proc/sys/vm/drop_caches", "w") as caches:
+                    caches.write('3')
```

Comment:
It would be better to place the drop cache in setup() so it will work for setting hugepages on numa nodes too.